skills_ml/algorithms/embedding/models.py

```diff
@@ -0,0 +1,137 @@
+"""Embedding model class inherited the interface from gensim"""
+from skills_ml.storage import Store, FSStore
+from skills_ml.algorithms.embedding.base import Base2VecModel
+
+from gensim.models import Doc2Vec, Word2Vec
+
+import numpy as np
+import pickle
+import logging
+
+class Word2VecModel(Word2Vec, Base2VecModel):
```

Comment:
It looks like a lot of this functionality could get pushed into the base class, while still being overridable by subclasses.

I started looking at the base class with the goal of figuring out: what is its purpose? At first glance, the base class looks like it could fit anything that has a store Object, but implies that the thing being stored is a model. All of these operations are defined by the subclasses. Makes sense. Then I realized the specific methods that needed to be implemented (load_model/infer_vector) are Doc2Vec/Word2Vec methods, so the name makes sense.

Then I look at the subclasses. I see that infer_vector is in one, and not the other, which implies that Doc2Vec implements that itself but Word2Vec needs to. This makes sense.

It's the other stuff that makes sense to me to push to the base class. The interaction with the storage appears to be the same in both, so why not make it the default? Doc2VecModel would then become tiny.